frontend/src/scenes/insights/InsightMetadata/InsightMetadata.tsx

```diff
@@ -0,0 +1,182 @@
+    const property = 'name'
+    const logic = insightMetadataLogic({ insight: { [property]: insight?.[property] } })
+    const { editableProps } = useValues(logic)
+    const { setInsightMetadata, saveInsightMetadata, cancelInsightMetadata, showEditMode } = useActions(logic)
+    const placeholder = insight[property] ?? `Insight #${insight.id ?? '...'}`
+
+    return (
+        <div className={createInsightInputClassName('title', isEditable)} data-attr="insight-title">
+            {isEditable ? (
+                editableProps.has(property) ? (
+                    <Input
+                        placeholder={`Insight #${insight.id ?? '...'}`}
+                        defaultValue={insight[property] ?? ''}
+                        size="large"
+                        onChange={(e) => {
+                            setInsightMetadata({ [property]: e.target.value })
+                        }}
+                        onKeyDown={(e) => {
+                            if (e.key === 'Enter') {
+                                saveInsightMetadata(property)
+                            }
+                        }}
+                        tabIndex={0}
+                        suffix={
+                            <>
+                                <Button
+                                    className="btn-cancel"
+                                    size="small"
+                                    onClick={() => cancelInsightMetadata(property)}
+                                >
+                                    Cancel
+                                </Button>
+                                <Button
+                                    className="ml-025"
+                                    type="primary"
+                                    size="small"
+                                    onClick={() => saveInsightMetadata(property)}
+                                >
+                                    Done
+                                </Button>
+                            </>
+                        }
+                    />
+                ) : (
+                    <>
+                        <span className="title">{placeholder}</span>
+                        <Button
+                            type="link"
+                            onClick={() => {
+                                showEditMode(property)
+                            }}
+                            className="btn-edit"
+                            data-attr={`edit-prop-${property}`}
+                            title={`Edit ${property}`}
+                        >
+                            <EditOutlined />
+                        </Button>
+                    </>
+                )
+            ) : (
+                <span className="title">{placeholder}</span>
+            )}
+        </div>
+    )
+}
+
+function Description({ insight, isEditable = false }: MetadataProps): JSX.Element | null {
```

Comment:
I wonder if there's value in refactoring to extract an `EditableTextField` component. It would reduce duplication of the idea here but could also be lifted out as a component

Reply:
Hmm, in my first iterations I tried to refactor this part out, but I found it difficult and even more confusing to read especially with the api differences between `Input.TextArea` and `Input`. It'd be really nice if `Input.TextArea` supported the `suffix` prop so that I don't need this hacky wrapper workaround, but until then I think it'd be better to leave this for now.